source/common/common/utility.h

```diff
@@ -288,4 +301,57 @@ class RegexUtil {
                                std::regex::flag_type flags = std::regex::optimize);
 };
 
+/**
+ * Maintains sets of numeric intervals. As new intervals are added, existing ones in the
+ * set are combined so that no overlapping intervals remain in the representation.
+ *
+ * Value can be any type that is comparable with <, ==, and >.
+ */
+template <typename Value> class IntervalSetImpl : public IntervalSet<Value> {
+public:
+  typedef typename IntervalSet<Value>::Interval Interval;
```

Comment:
A short comment reminder on this indicating that an `IntervalSet<Value>::Interval` is a pair of Values would have helped me read this.

Reply:
Done